addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,504 @@
+                    ? previous.querySelector(
+                          ".o_embedded_toggle_button:has(.fa-caret-right) + div [data-embedded-editable='title'], [data-embedded-editable='content']"
+                      )
+                    : previous,
+                isTextNode
+            );
+        }
+        this.dependencies.selection.setSelection({
+            anchorNode,
+            anchorOffset: Math.min(selection.anchorOffset, anchorNode.textContent.length),
+        });
+    }
+
+    _arrowDown(selection) {
+        const container = closestElement(selection.anchorNode, "[data-embedded-editable]");
+        let next;
+        let anchorNode;
+        if (
+            container.dataset.embeddedEditable === "title" &&
+            container.parentElement.previousSibling.querySelector(".fa-caret-down")
+        ) {
+            next = container.parentElement.parentElement.nextSibling;
+            anchorNode = firstLeaf(
+                next,
+                (node) => closestElement(node, "[data-embedded-editable]") && isTextNode(node)
+            );
+        } else {
+            next = closestElement(
+                container,
+                (element) => element.dataset.embedded === "toggle" && element.nextSibling
+            ).nextSibling;
+            anchorNode = firstLeaf(
+                next.dataset.embedded === "toggle"
+                    ? next.querySelector("[data-embedded-editable='title']")
+                    : next,
+                isTextNode
+            );
+        }
+        this.dependencies.selection.setSelection({
+            anchorNode,
+            anchorOffset: Math.min(selection.anchorOffset, anchorNode.textContent.length),
+        });
+    }
+
+    showPowerButtons(selection) {
+        return selection.isCollapsed && !closestElement(selection.anchorNode, titleSelector);
+    }
+
+    setupNewToggle({ name, env }) {
+        if (name === "toggle") {
+            Object.assign(env, {
+                editorShared: {
+                    preserveSelection: this.dependencies.selection.preserveSelection,
+                },
+            });
+        }
+    }
+
+    normalize(element) {
+        const toggleNodes = element.querySelectorAll("[data-embedded='toggle']");
+        for (const toggleNode of toggleNodes) {
+            const target = toggleNode.querySelector("[data-embedded-editable]:empty");
+            if (target) {
+                const newParagraph = this.document.createElement("p");
+                newParagraph.appendChild(this.document.createElement("br"));
+                target.replaceChildren(newParagraph);
+            }
+        }
+    }
+
+    beforePaste(selection, ev) {
+        const { anchorNode } = selection;
+        const closestToggleTitle = closestElement(anchorNode, titleSelector);
+        if (!closestToggleTitle) {
+            return;
+        }
+        const fragmentToCheck = parseHTML(this.document, ev.clipboardData.getData("text/html"));
```

Comment:
From my understanding, letting the commands available inside the title made sense.
With the toggle you could have a details section for each command if needed.
Blocking the pasting is another thing, as this prevents users to introduce unsafe elements to the article.
Would be glad to launch the debate on this one, as blocking the commands would require some more thinking since every user_commands would need to be passed through a check.
This check would be done through a resource that would group multiple functions.
Each function would take as arguments the selection and the command.